src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Patterns.cs

```diff
@@ -52,12 +52,12 @@ private BoundExpression LowerConstantPattern(BoundConstantPattern pattern, Bound
 
         private BoundExpression LowerDeclarationPattern(BoundDeclarationPattern pattern, BoundExpression input)
         {
-            Debug.Assert(pattern.IsVar || pattern.LocalSymbol.Type == pattern.DeclaredType.Type);
+            Debug.Assert(pattern.IsVar || pattern.Variable.GetTypeOrReturnType() == pattern.DeclaredType.Type);
```

Comment:
>pattern.IsVar  [](start = 25, length = 14)

What is the purpose of ```IsVar``` condition? Shouldn't Variable have type assigned before we get here?